topostats/scars.py

```diff
@@ -333,32 +333,31 @@ def _remove_marked_scars(img: np.ndarray, scar_mask: np.ndarray) -> None:
     None
     """
 
-    for row in range(img.shape[0]):
-        for col in range(img.shape[1]):
-            if scar_mask[row, col] == 1.0:
-                # Scar detected
-                # Find out how tall it is
-                scar_width = 1
-                while True:
-                    # Exit if reach the bottom of the image
-                    if row + scar_width >= img.shape[0]:
-                        break
-                    # If pixel is marked, increase scar width
-                    if scar_mask[row + scar_width, col] == 1.0:
-                        scar_width += 1
-                    else:
-                        break
-
-                above = img[row - 1, col]
-                below = img[row + scar_width, col]
-                k = scar_width
-                while k:
-                    # Linearly interpolate
-                    interp_val = (k / (scar_width + 1)) * below + (1 - (k / (scar_width + 1))) * above
-                    img[row + k - 1, col] = interp_val
-                    scar_mask[row + k - 1, col] = 0.0
-                    k -= 1
-                    LOGGER.debug("Scar removed")
+    for row, col in np.ndindex(img.shape):
+        if scar_mask[row, col] == 1.0:
+            # Scar detected
+            # Find out how tall it is
+            scar_width = 1
+            while True:
```

Comment:
@ns-rse I thought that in making the code a bit more verbose with the loop like that, it would make it more clear, sorry. I have made a change that is based off of your suggestion. I had to edit it to make it work but I think it's in the same spirit?

Let me know if this change is the kind of thing you were intending.